ui-v2/src/components/concurrency/global-concurrency-view/create-or-edit-limit-dialog/use-create-or-edit-limit-dialog.ts

```diff
@@ -0,0 +1,121 @@
+import {
+	GlobalConcurrencyLimit,
+	useCreateGlobalConcurrencyLimit,
+	useUpdateGlobalConcurrencyLimit,
+} from "@/hooks/global-concurrency-limits";
+import { useToast } from "@/hooks/use-toast";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+const formSchema = z.object({
+	active: z.boolean().default(true),
+	/** Coerce to solve common issue of transforming a string number to a number type */
+	denied_slots: z.number().default(0).or(z.string()).pipe(z.coerce.number()),
+	/** Coerce to solve common issue of transforming a string number to a number type */
+	limit: z.number().default(0).or(z.string()).pipe(z.coerce.number()),
+	name: z
+		.string()
+		.min(2, { message: "Name must be at least 2 characters" })
+		.default(""),
+	/** Coerce to solve common issue of transforming a string number to a number type */
+	slot_decay_per_second: z
+		.number()
+		.default(0)
+		.or(z.string())
+		.pipe(z.coerce.number()),
+	/** Additional fields post creation. Coerce to solve common issue of transforming a string number to a number type  */
+	active_slots: z.number().default(0).or(z.string()).pipe(z.coerce.number()),
+});
+
+const DEFAULT_VALUES = {
+	active: true,
+	name: "",
+	limit: 0,
+	slot_decay_per_second: 0,
+	denied_slots: 0,
+	active_slots: 0,
+} as const;
+
+type UseCreateOrEditLimitDialog = {
+	/** Limit to edit. Pass undefined if creating a new limit */
+	limitToUpdate: GlobalConcurrencyLimit | undefined;
+	/** Callback after hitting Save or Update */
+	onSubmit: () => void;
+};
+
+export const useCreateOrEditLimitDialog = ({
```

Comment:
I think it makes sense for this to be `useCreateOrEditLimitForm` since it appears to only handle form matters and not any dialog state.

Having this logic separate really cleans up the component code!
```suggestion
export const useCreateOrEditLimitForm = ({
```

Reply:
updated